java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -125,4 +139,16 @@ private static boolean hasPublicModifier(MethodTree methodTree) {
     return ModifiersUtils.hasModifier(methodTree.modifiers(), Modifier.PUBLIC);
   }
 
+  private static class SpringBootVisitor extends BaseTreeVisitor {
+
+    private static final MethodMatcher SPRING_BOOT = MethodMatcher.create().typeDefinition("org.springframework.boot.SpringApplication").name("run").withAnyParameters();
+    boolean springBootDetected;
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree tree) {
+      if (SPRING_BOOT.matches(tree)) {
```

Comment:
What if  `run` method is called within an anonymous class ? do we want to raise an issue ? This probably deserves a testcase.

```
void foo() {
   new Anonymous() {
        void plop() {
                SpringApplication.run(); // shall we raise an issue or not ?
         }
  }
}